.github/workflows/clean_workflow.yml

```
@@ -0,0 +1,118 @@
+---
+name: Clean Out Branches
+
+on:
+  workflow_call:
+    inputs:
+      branch:
+        required: true
+        type: string
+    secrets:
+      token:
+        required: true
+      DEV_KUBE_CONFIG:
+        required: true
+
+jobs:
+  create_artifacts_repos:
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        repos: 
+         - schulcloud-server
+         - schulcloud-client
+         - nuxt-client
+         - superhero-dashboard
+         - schulcloud-calendar
+         - edtrio
+         - schulcloud-editor 
+         - antivirus_check_service
+         - dof_app_deploy
+    steps:
+      - run: |
+          echo "git_ref_name=${{ inputs.branch }}" >> $GITHUB_ENV
+          echo git_ref_name ${{ inputs.branch }}
+      - uses: actions/checkout@v2
+        with:
+          repository: hpi-schul-cloud/${{ matrix.repos }}
+          token: ${{ secrets.token }}
+          path: ${{ matrix.repos }}
+          fetch-depth: 0
+      - run: git checkout ${{ env.git_ref_name }}
+        working-directory: ${{github.workspace }}/${{ matrix.repos }}
+      - name: Add Flag for existing branch
+        if: ${{ success() }}
+        working-directory: ${{github.workspace }}/${{ matrix.repos }}
+        shell: bash
+        run: |
+          mkdir -pv ansible/group_vars/all
+          echo "BRANCH_EXIST: true" > ansible/group_vars/all/${{ matrix.repos }}.yml
+      - run: tar -cf ${{ matrix.repos }}.tar ansible
+        if: ${{ always() }}
+        working-directory: ${{github.workspace }}/${{ matrix.repos }}
+      - uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: ${{ matrix.repos }}
+          path: ${{github.workspace }}/${{ matrix.repos }}/${{ matrix.repos }}.tar
+          
+  create_artifacts_workspaces:
+    if: ${{ always() }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tennens: [ brb, default, nbc ]
+    steps:
+      - run: |
+          echo "git_ref_name=${{ inputs.branch }}" >> $GITHUB_ENV
+          echo git_ref_name ${{ inputs.branch }}
+      - name: SHA Stuf
```

```suggestion
      - name: SHA Stuff
```
Maybe we can find a better name here? That's for the prefix gerneration for the domain, right?